python/cudf/cudf/core/column/column.py

```diff
@@ -1639,22 +1639,23 @@ def build_column(
         )
     elif isinstance(dtype, IntervalDtype):
         return cudf.core.column.IntervalColumn(
+            data=None,
+            size=size,  # type: ignore[arg-type]
```

Comment:
I assume eventually we'll want to align the type fully by deciding whether or not to allow None?

Reply:
Yes I'd prefer that. Like discussed in https://github.com/rapidsai/cudf/pull/16576#discussion_r1719171709 it would be good if this was always required to be an `int`